test/cypress/support/commands.js

```diff
@@ -133,18 +164,28 @@ Cypress.Commands.add('giCreateGroup', (name, {
   bypassUI = false
 } = {}) => {
   if (bypassUI) {
-    cy.window().its('sbp').then(async sbp => {
-      await sbp('gi.actions/group/createAndSwitch', {
-        data: {
-          name,
-          sharedValues,
-          mincomeAmount: mincome,
-          mincomeCurrency: 'USD',
-          ruleName,
-          ruleThreshold
-        }
+    cy.window().its('sbp').then(sbp => {
+      return new Promise(resolve => {
+        (async () => {
+          const message = await sbp('gi.actions/group/createAndSwitch', {
+            data: {
+              name,
+              sharedValues,
+              mincomeAmount: mincome,
+              mincomeCurrency: 'USD',
+              ruleName,
+              ruleThreshold
+            }
+          })
+
+          sbp('okTurtles.events/on', JOINED_GROUP, async ({ contractID }) => {
+            if (contractID === message.contractID()) {
+              await sbp('controller/router').push({ path: '/dashboard' }).catch(e => {})
+              resolve()
+            }
```

Comment:
We probably want to remove the event listener just before calling `resolve()` so that it doesn't get called again

Reply:
Yeah, nice catch.